src/nni_engine.hpp

```diff
@@ -132,30 +231,112 @@ class NNIEngine {
                                               const bool is_edge_on_left,
                                               const bool is_edge_leafward);
   // Based on given input NNIOperation, produces the two possible output NNIOperations
-  // and adds those results to the NNI Set (if results are not a member of the DAG).
+  // and adds those results to the NNI Set (if results are not a member of the DAG or
+  // NNI Set).
   void SafeAddOutputNNIsToAdjacentNNIs(const Bitset &parent_bitset,
                                        const Bitset &child_bitset,
                                        const bool is_edge_on_left);
-  // Remove all adjacent nnis from list.
-  void ClearAdjacentNNIs();
-  // Remove all accepted nnis from list.
-  void ClearAcceptedNNIs();
-  // Remove all rejected nnis from list.
-  void ClearRejectedNNIs();
+
+  // Add score to given NNI.
+  void AddScoreForNNI(const NNIOperation &nni, const double score);
+  // Get score for given NNI.
+  double GetScoreForNNI(const NNIOperation &nni) const;
+
+  // Remove all adjacent nnis from list. Option to keep and re-evaluate NNIs from
+  // previous pass.
+  void UpdateAdjacentNNIs(const bool reevaluate_rejected_nnis = false);
```

Comment:
Perhaps rename to something with Remove?

Reply:
Added this clarifying comment to what the function is doing.  I think that "Update" is appropriate, but tell me what you think.
  
  // Update adjacent NNIs at end of current sweep.  If not re-evaluating rejected NNIs,
  // then current adjacent NNIs are removed.  Then NNIs that are adjacent to last sweep's
  // accepted NNIs are added.